plugins/woocommerce/src/Internal/Admin/Settings/PaymentsProviders/WCCore.php

```diff
@@ -91,4 +91,25 @@ public function is_in_test_mode_onboarding( WC_Payment_Gateway $payment_gateway
 
 		return parent::is_in_test_mode_onboarding( $payment_gateway );
 	}
+
+
+	/**
+	 * Hide the "Deactivate" action for legacy PayPal Standard by omitting the plugin file.
+	 *
+	 * @param \WC_Payment_Gateway $payment_gateway
+	 * @return array
+	 */
+	public function get_plugin_details( \WC_Payment_Gateway $payment_gateway ): array {
+		$details = parent::get_plugin_details( $payment_gateway );
+
+		// Legacy PayPal Standard (gateway id 'paypal') lives in WooCommerce core.
+		// If we provide a plugin 'file', the UI will offer "Deactivate" for that plugin.
+		// Remove the file so the UI cannot render a Deactivate action for this row.
+		if ( \WC_Gateway_Paypal::ID === $payment_gateway->id ) {
```

Comment:
I believe we can do it even simpler. Since this provider class is for any WC core payment gateway, we can always provide an empty file, not just for PayPal. This way, we err on the safe side in case other core payment gateways appear in the future.